crates/store/src/db/transaction.rs

```diff
@@ -0,0 +1,41 @@
+use std::ops::{Deref, DerefMut};
+
+/// SQLite transaction wrapper for optional query plan rendering.
+pub struct Transaction<'conn> {
+    inner: rusqlite::Transaction<'conn>,
+}
+
+impl<'conn> Transaction<'conn> {
+    pub(super) fn new(inner: rusqlite::Transaction<'conn>) -> Self {
+        Self { inner }
+    }
+
+    #[inline]
+    pub fn commit(self) -> rusqlite::Result<()> {
+        self.inner.commit()
+    }
+}
+
+impl<'conn> Deref for Transaction<'conn> {
+    type Target = rusqlite::Transaction<'conn>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl DerefMut for Transaction<'_> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.inner
+    }
+}
+
+impl Transaction<'_> {
+    #[inline]
+    pub fn prepare_cached(&self, sql: &str) -> rusqlite::Result<rusqlite::CachedStatement<'_>> {
+        #[cfg(feature = "explain-query-plans")]
+        self.explain_query_plan(sql)?;
+
+        self.inner.prepare_cached(sql)
+    }
```

Comment:
I don't think there is a need to enable this on a production node. Or rather I don't think there will ever be a valid time to do so.

I think we can do this purely for tests which should cover all queries/statements.

I also think the check _should not_ print anything by default.

```suggestion
    pub fn prepare_cached(&self, sql: &str) -> rusqlite::Result<rusqlite::CachedStatement<'_>> {
       // We do this first so we get invalid sql errors here
       // instead of as a panic in the query plan check for tests.
       let stmt = self.inner.prepare_cached(sql)?;
    
        #[cfg(test)]
        self.check_query_plan(sql);

        Ok(stmt)
    }
```

where we only print the query plan tree on failure:

```rust
#[cfg(test)]
impl Transaction<'_> {
    /// Panics if the query plan contains a slow table scan.
    fn check_query_plan(&self, sql: &str) {
        let query_plan = self.query_plan(sql).expect("valid sql");

        assert!(query_plan.contains_scan().not(), "Query plan contains a table scan: \n{query_plan});
    }
}
```